src/marks/arrow.js

```diff
@@ -91,6 +87,23 @@ export class Arrow extends Mark {
             // wings, but that’s okay since vectors are usually small.)
             const headLength = Math.min(wingScale * sw(i), lineLength / 3);
 
+            // Maybe flip the bending, if a sweep order is applied.
+            const flip =
+              this.sweep == null
+                ? 1
+                : this.sweep === "order-x"
+                ? descending(x1, x2)
```

Comment:
This will return 0 when x1 == x2, which will result in a bend angle of 0 below, which I don’t think is correct?

Reply:
The use case I had in mind for *order-x* is when *y* is constant; in that case this test would be for links of length 0, and bending can be 0. 

However I hadn't really given much thought to the fact that *y* can be variable. In that case, I don't really know what the user intent might be by specifying *order-x* (instead of *order*, which always “bends” unless the link has length 0). The current code, which makes a straight line when the order is a tie on x, doesn't look wrong and might even be useful (?):
```
Plot.arrow([-2, -1, 0, 1, 2], {
  x1: Plot.identity,
  x2: (d) => d * 1.1,
  y1: 0,
  y2: 1,
  sweep: "order-x",
  bend: -20
}).plot()
```
![](https://github.com/observablehq/plot/assets/7001/3f28ad62-24e8-41ce-8d71-71b73df3fbb7)

Reply:
I think it’s okay if you document it.